tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4757,10 +4922,17 @@ class HLSLExternalSource : public ExternalSemaSource {
         ULE->getQualifier()->getKind() == NestedNameSpecifier::Namespace &&
         ULE->getQualifier()->getAsNamespace()->getName() == "vk";
 
+    const bool isDxNamespace =
+        ULE->getQualifier() &&
+        ULE->getQualifier()->getKind() == NestedNameSpecifier::Namespace &&
+        ULE->getQualifier()->getAsNamespace()->getName() == "dx";
+
     // Intrinsics live in the global namespace, so references to their names
     // should be either unqualified or '::'-prefixed.
-    // Exception: Vulkan-specific intrinsics live in the 'vk::' namespace.
-    if (isQualified && !isGlobalNamespace && !isVkNamespace) {
+    // Exceptions:
+    // - Vulkan-specific intrinsics live in the 'vk::' namespace.
+    // - DirectX-specific intrinsics live in the 'dx::' namespace.
+    if (isQualified && !isGlobalNamespace && !isVkNamespace && !isDxNamespace) {
       return false;
     }
```

Comment:
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
```suggestion
    if (isQualified && !isGlobalNamespace && !isVkNamespace && !isDxNamespace)
      return false;
```

Reply:
Applied in [ba11d4f](https://github.com/microsoft/DirectXShaderCompiler/pull/7097/commits/ba11d4f3ffde25c5945852099caf9e983b2deb58)